tests/functional/t/typevar_naming_style_default.py

```diff
@@ -0,0 +1,53 @@
+"""Test case for typevar-name-incorrect-variance with default settings"""
+# pylint: disable=too-few-public-methods
+
+from typing import TypeVar
+
+
+class MyClass:
+    """Class to be used to bound TypeVar's"""
+
+
+# PascalCase names with prefix
+GoodNameT = TypeVar("GoodNameT")
+_T = TypeVar("_T")
+_GoodNameT = TypeVar("_GoodNameT")
+__GoodNameT = TypeVar("__GoodNameT")
+GoodNameWithoutContra = TypeVar(  # [typevar-name-incorrect-variance]
+    "GoodNameWithoutContra", contravariant=True
+)
+GoodNameT_co = TypeVar("GoodNameT_co", covariant=True)
+GoodNameT_contra = TypeVar("GoodNameT_contra", contravariant=True)
+GoodBoundNameT = TypeVar("GoodBoundNameT", bound=MyClass)
+GoodBoundNameT_co = TypeVar(  # [typevar-name-incorrect-variance]
+    "GoodBoundNameT_co", bound=MyClass
+)
+GoodBoundNameT_contra = TypeVar(  # [typevar-name-incorrect-variance]
+    "GoodBoundNameT_contra", bound=MyClass
+)
+
+# PascalCase names without prefix
+AnyStr = TypeVar("AnyStr")
+BadNameWithoutContra = TypeVar(  # [typevar-name-incorrect-variance]
+    "BadNameWithoutContra", contravariant=True
+)
+BadName_co = TypeVar("BadName_co", covariant=True)
+BadName_contra = TypeVar("BadName_contra", contravariant=True)
+
+# camelCase names with prefix
+T_goodName = TypeVar("T_goodName")  # [invalid-name]
+T_goodNameWithoutContra = TypeVar(  # [invalid-name, typevar-name-incorrect-variance]]
+    "T_goodNameWithoutContra", contravariant=True
+)
+T_goodName_co = TypeVar("T_goodName_co", covariant=True)  # [invalid-name]
+T_goodName_contra = TypeVar("T_goodName_contra", contravariant=True)  # [invalid-name]
```

Comment:
These bad ones. It's probably best to remove the `T_` prefix too.

Reply:
https://github.com/PyCQA/pylint/pull/5221/commits/0bde758cc754cd5b81bbb526aa4d055cf33bdc97